### PR TITLE
10214 improving comments and adding tests to tree searcher

### LINKED ISSUE #10214: Improving comments and adding tests to TreeSearcher
TreeSearcher needs love...
We will add comment and tests

### PR DESCRIPTION
**Pull request by alesshosry**
Adding more tests + Enhancing class RBParseTreeSearcher

### CODE REVIEW

**Comment by Ducasse** [comment]
Thanks a lot Aless for your contribution. We will have a look. 
Having more tests is super nice. Thanks again.

File: src/AST-Core-Tests/RBParseTreeSearcherTest.class.st [hunk 1]
```diff
@@ -46,24 +46,487 @@ RBParseTreeSearcherTest >> testDynamicArrayWithStatementListPattern [
 
 	| dict |
 	searcher matches: '{`.@stmts.}' do: [ :aNode :answer |  dict:= searcher context ].
-	dict := searcher executeTree: (self parseExpression:'{ (1@2) . Color red . 3}').
+	dict := searcher executeTree: (self parseExpression:'{ (1@2) . Color red . 3 }').	
+	"self halt."
```
Comment (gcotelli) [inline src/AST-Core-Tests/RBParseTreeSearcherTest.class.st:50]: Remove this debugging code, even if it's commented.

File: src/AST-Core-Tests/RBParseTreeSearcherTest.class.st [hunk 2]
```diff
@@ -46,24 +46,487 @@ RBParseTreeSearcherTest >> testDynamicArrayWithStatementListPattern [
...
 
 ]
 
+{ #category : #'tests - to de refined' }
+RBParseTreeSearcherTest >> testSearchListComplex [
+
+	"		
+		- Here we looking for multiple lists in one match pattern at the same time. 
+		- Lists can be any receiver or args
+	"
```
Comment (gcotelli) [inline src/AST-Core-Tests/RBParseTreeSearcherTest.class.st]:
```suggestion
	"Here we are looking for multiple lists in one match pattern at the same time. 
	Lists can be any receiver or arguments"
```

File: src/AST-Core-Tests/RBParseTreeSearcherTest.class.st [hunk 3]
```diff
@@ -46,24 +46,487 @@ RBParseTreeSearcherTest >> testDynamicArrayWithStatementListPattern [
...
+		matches: '`@receiver assert: `@arg equals: true'
+		do: [ :aNode :answer | dict := searcher context ].
+
+	"so in below example, there is no match with the suggested pattern. Which is why, dict is nil "
+
+	searcher executeTree:
+		(self parseExpression: 'self assert: token isLiteralToken.').
+
+	self assert: dict isNil
+	"self deny: dict isNotNil"
```
Comment (gcotelli) [inline src/AST-Core-Tests/RBParseTreeSearcherTest.class.st]:
```suggestion
	"Here we are looking for multiple lists in one match pattern at the same time. 
	Lists can be any receiver or argument"

	searcher
		matches: '`@receiver assert: `@arg equals: true'
		do: [ :aNode :answer | self fail ].

	searcher executeTree:
		(self parseExpression: 'self assert: token isLiteralToken.').
```

**Review by gcotelli — CHANGES_REQUESTED** [review]
- Use proper names instead of `dict`
- Method comments are formatted in an strange way
- The cases when the searcher is not matching can be changed to do `self fail` in the do block instead of checking for nil (I'm proposing one of that changes in this review)
- There's a lot of almost repeated assertions, like
```
self
		assert:
		(dict at: (RBPatternVariableNode named: '`@arg')) formattedCode
		equals: 'offset + count'.
```
extract it to something receiving the context (dict) and the variable name, something like:
`assertCodeForVariableNodeNamed: name in: context equals: expectedCode`

**Comment by MarcusDenker** [comment]
This PR just needs a pass over the comments of the reviewer

**Comment by Ducasse** [comment]
AlessHo can you have a look?

**Comment by MarcusDenker** [comment]
@AlessHo

**Comment by AlessHo** [comment]
Hello, Okay

**Comment by alesshosry** [comment]
Hello,

I just saw the messages.
Ok, noted for the comments I will adjust them this week.
Thank you for your reviews.

**Pull request merged**